File: services/task-service/internal/core/domain/task.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// TaskStatus - перечисление для статусов задачи
type TaskStatus string

const (
	StatusPending   TaskStatus = "pending"
	StatusRunning   TaskStatus = "running"
	StatusCompleted TaskStatus = "completed"
	StatusFailed    TaskStatus = "failed"
)

// ResultSummary - структура для хранения сводной информации о результатах
// Использование map[string]interface{} делает ее гибкой для разных типов задач
type ResultSummary map[string]interface{}

// Task - основная доменная сущность
type Task struct {
	ID               uuid.UUID		`json:"id"`
	Name             string			`json:"name"`
	Type             string			`json:"type"`
	Status           TaskStatus		`json:"status"`
	ResultSummary    ResultSummary	`json:"result_summary"`
	CreatedAt        time.Time		`json:"created_at"`
	StartedAt        *time.Time 	`json:"started_at"`
	FinishedAt       *time.Time 	`json:"finished_at"`
	CreatedByUserID  uuid.UUID		`json:"created_by_user_id"`
	TargetObjectID 	 *uuid.UUID 	`json:"target_object_id,omitempty"` 
}

// NewTask - конструктор для создания новой задачи
func NewTask(name, taskType string, createdByUserID uuid.UUID) *Task {
	return &Task{
		ID:              uuid.New(),
		Name:            name,
		Type:            taskType,
		Status:          StatusPending, // Начальный статус
		CreatedAt:       time.Now().UTC(),
		CreatedByUserID: createdByUserID,
		ResultSummary: make(ResultSummary),
	}
}

File: ts_client/src/types/task.ts
// Описание одной задачи
export interface ITask {
    id: string;
    name: string;
    type: 'ACTUALIZE_BY_ID' | 'ACTUALIZE_ACTIVE' | 'FIND_NEW';
    status: 'pending' | 'running' | 'completed' | 'failed';
    created_at: string;
    started_at?: string;
    finished_at?: string;
    
    // Результаты (обновляются в реальном времени)
    result_summary?: {
        expected_results_count: number; // Сколько всего ожидаем (100%)
        total_processed: number;        // Сколько сделали
        created: number;
        updated: number;
        archived: number;
        new_links_found: number;
        id?: string;
    };
    created_by_user_id: string;
}

// Ответ списка задач (с пагинацией, как у вас в JSON)
export interface ITasksResponse {
    data: ITask[];
    total: number;
    page: number;
    perPage: number;
}